env/staging/create_metrics_lambda/lambda/create_metric.js

```diff
@@ -21,31 +21,69 @@ exports.handler = async (event, context) => {
     // expire after 24 hours
     const ttl = (Math.floor(Date.now()/1000) + 86400).toString();
 
-    const params = {
-        TableName: process.env.TABLE_NAME,
-        Item : {
-            "uuid": {
-                S: uuidv4(),
-            },
-            "expdate" : {
-                N: ttl,
-            },
-            "raw": {
-                S: event.body,
-            },
-        },
-    };
-
+    const eventBody = event.body;
+    
     try {
+        // The maximum item size in DynamoDB is 400 KB
+        const payloadLength = new TextEncoder().encode(event.body.payload).length;
+        if (payloadLength > process.env.SPLIT_THRESHOLD){
+            console.info(`Large payload being split; size: ${payloadLength}`);
+            const results = splitPayload(event.body.payload);
 
-        await dynamodb.putItem(params).promise();
+            results.forEach((result) => {
+                result.forEach((r) => {
+                    eventBody.payload = r
+                    writePayload(eventBody, ttl);
+                });
+            });
+        }else{
+            await writePayload(event.body, ttl);
+        }
         transactionStatus.statusCode = 200;
         transactionStatus.body = JSON.stringify({ "status": "RECORD CREATED" });
     } catch (err) {
-        console.error(`Upload faile ${err}`);
+        console.error(`Upload failed ${err}`);
         transactionStatus.statusCode = 500;
         transactionStatus.body= JSON.stringify({ "status" : "UPLOAD FAILED" });
     }
 
     return transactionStatus;
 };
+
+// Recursively splits the payload in half until all chunks are below the limit
+const splitPayload = (payload) => {
+  const results = [];
+  
+  const middle = payload.length / 2; // if it's odd, it'll round down
+  const left = payload.slice(0, middle);
+  const right = payload.slice(middle, payload.length);
+  
+  if (new TextEncoder().encode(left).length > process.env.SPLIT_THRESHOLD){
+      results.push(splitPayload(left));
+      results.push(splitPayload(right));
+      return results;
```

Comment:
Can we do this non-recursively so we end up with a single array to iterate over?

Something like (untested)
```javascript
let results = [];
chunk_size = payload.length;
while (true) {
  const left = payload.slice(0, chunk_size);
  if (new TextEncoder().encode(left).length < process.env.SPLIT_THRESHOLD) {
   break;
  }

  chunk_size /= 2;
  // Can't go smaller then a single element chunk
  if (chunk_size <= 1) {
    break;
  }
}

for (let i = 0; i < payload.length; i += chunk_size) {
  results.push(payload[i, i + chunk_size]);
}
return results;
```

This is assuming that a given metric is about the same-ish size as any other metric.....

Reply:
tested recursive and this approach and this is ~10 times faster since it pre-calculates chunk size and does it in one try and doesnt have too much array manipulation. Thanks for the suggestion!